pedpy/methods/method_utils.py

```diff
@@ -1081,3 +1105,169 @@ def _check_crossing_in_frame_range(
     crossed[column_name] = crossed[column_name] == "both"
     crossed = crossed[crossed[column_name]]
     return crossed
+
+
+def _compute_orthogonal_speed_in_relation_to_proprotion(
+    group: pd.DataFrame, measurement_line: MeasurementLine
+) -> pd.DataFrame:
+    """Calculates the speed orthogonal to the line times partial line length.
+
+    group is a DataFrameGroupBy containing the columns
+        'v_x', 'v_y' and 'polygon'.
+    """
+    normal_vector = measurement_line.normal_vector()
+    return (
+        group[V_X_COL] * normal_vector[0] + group[V_Y_COL] * normal_vector[1]
+    ) * _compute_partial_line_length(group[POLYGON_COL], measurement_line)
+
+
+def _compute_partial_line_length(
+    polygon: shapely.Polygon, measurement_line: MeasurementLine
+) -> float:
+    """Calculates the fraction of the length that is intersected by the polygon.
+
+    .
+    """
+    line = measurement_line.line
+    return shapely.length(shapely.intersection(polygon, line)) / shapely.length(
+        line
+    )
+
+
+def _apply_lambda_for_intersecting_frames(
+    *,
+    individual_voronoi_polygons: pd.DataFrame,
+    measurement_line: MeasurementLine,
+    species: pd.DataFrame,
+    lambda_for_group: typing.Callable[
+        [pd.DataFrame, MeasurementLine], pd.DataFrame
+    ],
+    column_id_sp1: str,
+    column_id_sp2: str,
+    individual_speed: pd.DataFrame = None,
+) -> pd.DataFrame:
+    """Applies lambda for both species for frames where Polygon intersects line.
+
+    lambda_for_group is called with a group containing
+     the data of one species and a Measurement Line.
```

Comment:
A proper docstring would be nice here too, this is used at multiple places and may be reused by other methods as well